### PR TITLE
MARP-434 Authentication SSH & Multi connection

### PR DESCRIPTION
**Pull request by anh-bolt**
https://1ivy.atlassian.net/browse/MARP-434
- update JSch to com.github.mwiede:jsch:0.2.19 (which support for loading new OPENSSH format key)
- add authentication with SSH key pair
- support multi connection with "sftpName" subset global variables

### CODE REVIEW

File: sftp-connector-demo/src/com/axonivy/connector/sftp/demo/Constants.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.axonivy.connector.sftp.demo;
+
+public class Constants {
+	public static final String DUMMY = "dummy";
```
Comment (majeed-hm) [inline sftp-connector-demo/src/com/axonivy/connector/sftp/demo/Constants.java:4]: Can you change the DUMMY to something meaningfull like TEST_SFTP_SERVER_NAME?

File: sftp-connector-demo/src_hd/com/axonivy/connector/sftp/demo/SftpClientDemo/SftpClientDemoData.ivyClass [hunk 2]
```diff
@@ -1,5 +1,7 @@
 SftpClientDemoData #class
 com.axonivy.connector.sftp.demo.SftpClientDemo #namespace
+sftpName String #field
```
Comment (majeed-hm) [inline sftp-connector-demo/src_hd/com/axonivy/connector/sftp/demo/SftpClientDemo/SftpClientDemoData.ivyClass:3]: Can you change sftpName to sftpServerName? to be more clear I think

File: sftp-connector/src/com/axonivy/connector/sftp/service/SftpClientService.java [hunk 3]
```diff
@@ -18,18 +20,29 @@
...
+
+	private static final String SFTP_VAR = "com.axonivy.connector.sftp.server";
+	private static final String HOST_VAR = "host";
+	private static final String PORT_VAR = "port";
+	private static final String SECRET_SSHPASSPHRASE_VAR = "secret.sshpassphrase";
+	private static final String SECRET_SSHKEY_VAR = "secret.sshkey";
+	private static final String AUTH_VAR = "auth";
+	private static final String PASSWORD_VAR = "password";
+	private static final String USERNAME_VAR = "username";
+	private static final String PASSWORD = "password";
```
Comment (majeed-hm) [inline sftp-connector/src/com/axonivy/connector/sftp/service/SftpClientService.java:44]: Can you rename PASSWORD to AUTH_METHOD? or better list the authentication methods in a proper Enum

File: sftp-connector-test/src_test/com/axonivy/connector/sftp/test/SftpProcessTest.java [hunk 4]
```diff
@@ -41,18 +41,19 @@ public class SftpProcessTest {
 	private static final BpmProcess TEST_UPLOAD_FILE_PROCESS = BpmProcess.path("Sftp/SftpUploadFile");
 	private static final BpmProcess TEST_DOWNLOAD_FILE_PROCESS = BpmProcess.path("Sftp/SftpDownloadFile");
 
+	private static final String TEST_SFTP_NAME = "dummy";
```
Comment (majeed-hm) [inline sftp-connector-test/src_test/com/axonivy/connector/sftp/test/SftpProcessTest.java:44]: Rename TEST_SFTP_NAME to TEST_SFTP_SERVER_NAME

**Review by majeed-hm — COMMENTED** [review]
Hi @anh-bolt ,
I finished my code review. can you please take a look? and feel free to ask me for any clarification.

**Review by majeed-hm — APPROVED** [review]

File: sftp-connector-product/README.md [hunk 5]
```diff
@@ -53,28 +53,66 @@ Before starting the demo, please make sure to have an SSH/SFTP server on your co
...
+         # The password to the SFTP server
+         password: ''
+
+         # The port number to the SFTP server
+         port: 22
+
+         # The username to the SFTP server
+         username: 'usr'
+       
+         # The ssh key string to SFTP server
```
Comment (linhpd-axonivy) [inline sftp-connector-product/README.md:105]: These configurations do not match the new variables.

**Review by linhpd-axonivy — APPROVED** [review]
LGTM

**Pull request merged**